clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -889,11 +929,20 @@ scheduler_tmpl_t<RecordType, ReaderType>::next_record(int output_ordinal,
                 record = **input->reader;
             }
         }
+        bool need_new_input = false;
+        if (options_.mapping == MAP_TO_ANY_OUTPUT &&
+            options_.quantum_unit == QUANTUM_INSTRUCTIONS &&
+            record_type_is_instr(record)) {
+            // TODO i#5843: We also want to swap on blocking syscalls.
+            ++input->instrs_in_quantum;
+            if (input->instrs_in_quantum > options_.quantum_duration)
+                need_new_input = true;
+        }
         if (options_.deps == DEPENDENCY_TIMESTAMPS &&
```

Comment:
This switches input on each timestamp marker? Even if the thread continues on the same core in the trace?

Reply:
I believe you're thinking of MAP_TO_RECORDED_OUTPUT which is not implemented for other than a single output (for analyzer serial mode): re-creating the recorded schedule on >1 output streams is not yet implemented.